dataprofiler/profilers/profile_builder.py

```diff
@@ -634,8 +695,13 @@ def _update_row_statistics(self, data, sample_ids=None):
                 null_rows = null_rows.intersection(null_row_indices)
                 null_in_row_count = null_in_row_count.union(null_row_indices)
 
-        self.row_has_null_count = len(null_in_row_count)
-        self.row_is_null_count = len(null_rows)
+        # If sample_ids provided, increment since that means only new data read
+        if sample_ids is not None:
+            self.row_has_null_count += len(null_in_row_count)
+            self.row_is_null_count += len(null_rows)
+        else:
+            self.row_has_null_count = len(null_in_row_count)
+            self.row_is_null_count = len(null_rows)
```

Comment:
```
if sample_ids is None:
     self.row_has_null_count = 0
     self.row_is_null_count = 0
self.row_has_null_count += len(null_in_row_count)
self.row_is_null_count += len(null_rows)
```

Reply:
I'm not following why this is necessary? This should be recalculated every time in the for loop above?

Reply:
@JGSweets I struggled to explain the solution while you were away, please revisit this and clarify.

Reply:
Basically, the following code would cause a problem if we just "recalculated" it unless we alter `_update_base_stats` to take in `sample_ids`. This would require collating all base_stats prior to updating them.
However, in the current case, if sample_ids exist in this function (because min_true_samples > 0), the following code will only intersect the data that was calculated in `update_profile` (unless you maintained the list fo sample_ids between updates).
```python
            if sample_ids is not None:
                # If sample ids provided, only consider nulls in rows that
                # were fully sampled
                null_row_indices = null_row_indices.intersection(
                    data.index[sample_ids[:self._min_col_samples_used]])
```
It would not be ideal to maintain the sample_ids across all updates or between merges of data.

Example:
First profile (`profiler = dp.Profiler(data1, min_true_samples=4)`):
assume len(data1) == 10
sample_ids = 2,6,8,9
indexes = 2,6,8,9
Null indexes == 2,6
Update profile (profiler.update_profile(data2):
assume len(data2) == 10
sample_ids = 1,3,4,9
indexes = 1,3,4,9 => (updated bc overlap) => (11, 13, 14, 19)
Null indexes = 3,4 => (updated bc overlap) => 13, 14

During `update_row_statistics` the following would occur in the current scenario
{2,6,13,14}.intersect((11, 13, 14, 19)) => {13,14}

if we kept the original code, 
```python
self.row_has_null_count = 2
self.row_is_null_count = 2 
```
the new code instead does
```python
# after first profile self.row_has_null_count = self.row_is_null_count = 2
self.row_has_null_count += 2
self.row_is_null_count += 2
```
Hence why we now do:
```python
        if sample_ids is not None:
            self.row_has_null_count += len(null_in_row_count)
            self.row_is_null_count += len(null_rows)
        else:
            self.row_has_null_count = len(null_in_row_count)
            self.row_is_null_count = len(null_rows)
```

Reply:
> ```
> if sample_ids is None:
>      self.row_has_null_count = 0
>      self.row_is_null_count = 0
> self.row_has_null_count += len(null_in_row_count)
> self.row_is_null_count += len(null_rows)
> ```
we could do that, i find it a little strange that we would set something to 0 and then update, but i do see the logic.

Reply:
> Basically, the following code would cause a problem if we just "recalculated" it unless we alter _update_base_stats to take in sample_ids. This would require collating all base_stats prior to updating them.

I see the issue, but I don't know if this is the appropriate solution. 

> {2,6,13,14}.intersect((11, 13, 14, 19)) => {13,14} # only when sample_ids is not None

This is a different problem that needs to be addressed. Shouldn't each update be isolated, i.e. 

    First profile (profiler = dp.Profiler(data1, min_true_samples=4)):
    assume len(data1) == 10
    sample_ids = 2,6,8,9
    indexes = 2,6,8,9
    Null indexes == 2,6

    Update profile (profiler.update_profile(data2)):
    assume len(data2) == 10
    sample_ids = 1,3,4,9
    indexes = 1,3,4,9 => (updated bc overlap) => (11, 13, 14, 19)
    Null indexes = 3,4 => (updated bc overlap) => 13, 14

    {2, 3}.union({13,14}.intersect((11, 13, 14, 19))) => {2, 3, 13,14} # only when sample_ids is not None

This should be the union of all the values (which would be the correct calculation):
```
self.row_has_null_count = len(null_in_row_count)
 self.row_is_null_count = len(null_rows)
```

Reply:
Ideally, `{13,14}.intersect((11, 13, 14, 19)) => {13, 14}` is calculated with base_stats

followed by a merge function which does:
```python
null_indexes = {2, 6}.union({13, 14})
self.row_has_null_count += len(updated_null_in_row_count)
self.row_is_null_count += len(updated_null_rows)
```
which would mirror what occurs in the lower level profilers of calculating all stats on the new dataset being added, followed by a merge function.

The problem is the redesign required to achieve this becomes complicated. We need to do the calc with `base_stats`, but this means compiling all base stats until all are processed bc of `sample_ids` and min_true_samples. Feasible, but would be what I see currently as the new technical requirement.

However, the current state is no different than the previous code despite an if statement. I would suggest that any such change should occur in a subsequent PR so that users can have the fix in now for the current errors in Nulls.